File: user-sites/user_3032/bf.py
#!/usr/bin/env python3
import sys

def run_bf(code):
    mem = [0] * 30000
    p = ip = 0
    loops = []
    output = []
    while ip < len(code):
        c = code[ip]
        if c == '>': p = (p + 1) % 30000
        elif c == '<': p = (p - 1) % 30000
        elif c == '+': mem[p] = (mem[p] + 1) & 255
        elif c == '-': mem[p] = (mem[p] - 1) & 255
        elif c == '.': output.append(chr(mem[p]))
        elif c == ',': pass  # no input
        elif c == '[':
            if not mem[p]:
                depth = 1
                while depth:
                    ip += 1
                    depth += (code[ip] == '[') - (code[ip] == ']')
            else:
                loops.append(ip)
        elif c == ']':
            if mem[p]:
                ip = loops[-1]
            else:
                loops.pop()
        ip += 1
    return ''.join(output)

if __name__ == '__main__':
    if len(sys.argv) < 2:
        print("Usage: bf.py <code>", file=sys.stderr)
        sys.exit(1)
    print(run_bf(sys.argv[1]), end='')

File: user-sites/user_3032/bf.js
#!/usr/bin/env node
const code = process.argv[2];
if (!code) { console.error("Usage: bf.js <code>"); process.exit(1); }
const mem = new Uint8Array(30000);
let p = 0, ip = 0;
const loops = [];
while (ip < code.length) {
    const c = code[ip];
    switch (c) {
        case '>': p = (p + 1) % 30000; break;
        case '<': p = (p - 1 + 30000) % 30000; break;
        case '+': mem[p]++; break;
        case '-': mem[p]--; break;
        case '.': process.stdout.write(String.fromCharCode(mem[p])); break;
        case ',': break; // no input in this version
        case '[': if (!mem[p]) { let d = 1; while (d) { ip++; d += (code[ip] === '[') - (code[ip] === ']'); } } else loops.push(ip); break;
        case ']': if (mem[p]) ip = loops[loops.length - 1]; else loops.pop(); break;
    }
    ip++;
}
